### PR TITLE
Add "taocpp_json" package

### PR DESCRIPTION
**Pull request by Bjoe**
* I've followed [this guide](https://docs.hunter.sh/en/latest/creating-new/create/cmake.html)
  step by step carefully. **Yes**

* I've tested this package remotely and have excluded all broken builds.
  Here is the links to the Travis/AppVeyor with status "All passed":

  MacOS/iOS:
  * `fatal error: 'optional' file not found` -> https://travis-ci.org/Bjoe/hunter/jobs/550113802#L687
  * `fatal error: 'optional' file not found` -> https://travis-ci.org/Bjoe/hunter/jobs/550113803#L958
  * `fatal error: 'optional' file not found` -> https://travis-ci.org/Bjoe/hunter/jobs/550113804#L985

 Windows vs-14-2015-sdk-8-1
  * Fails with fatal error C1083: Cannot open include file: 'optional': No such file or directory
    https://ci.appveyor.com/project/Bjoe/hunter/build/job/29dlf8svrklf3aal#L655

* I have submitted CI configs to https://github.com/ingenue/hunter targeting `pkg.template` branch,
  see this merged pull request https://github.com/ingenue/hunter/pull/375

### CODE REVIEW

**Comment by Bjoe** [comment]
Sorry, I didn't understand what is wrong here https://travis-ci.org/ruslo/hunter/builds/551741629#L2733 can somebody give me a hint?

**Comment by ruslo** [comment]
> Sorry, I didn't understand what is wrong here

It's a spell check.

Do

```
.. spelling::

    taocpp
    json
```

Instead of
```
.. spelling::

    taocpp-json
```

File: cmake/projects/taocpp-json/hunter.cmake [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+include(hunter_cacheable)
+include(hunter_cmake_args)
+include(hunter_download)
+include(hunter_pick_scheme)
+
+hunter_add_version(
+    PACKAGE_NAME
+    taocpp-json
+    VERSION
+    1.0.0-beta.11-9163
```
Comment (ruslo) [inline cmake/projects/taocpp-json/hunter.cmake]: What 9163 means?

```
> git describe --tags e0895587
1.0.0-beta.11-645-ge0895587
```

So I expect something like `1.0.0-beta.11-e0895587`, see
- https://docs.hunter.sh/en/latest/creating-new/patch.html.

Comment (Bjoe) [reply cmake/projects/taocpp-json/hunter.cmake]: Yep, absolutely correct. This was also my idea ... I copied/used the SHA1 :-(. Is fixed.

**Comment by ruslo** [comment]
Testing:
- https://ci.appveyor.com/project/ingenue/hunter/builds/25685944
- https://travis-ci.org/ingenue/hunter/builds/553187285?utm_source=github_status&utm_medium=notification

**Comment by ruslo** [comment]
Released: https://github.com/ruslo/hunter/releases/tag/v0.23.198

**Pull request merged**